frontend/src/metabase/lib/urls/dashboards.ts

```diff
@@ -32,8 +32,13 @@ export function dashboard(
   }
 }
 
-export function publicDashboard(uuid: string) {
-  const siteUrl = MetabaseSettings.get("site-url");
+export function publicDashboard({
+  uuid,
+  siteUrl = MetabaseSettings.get("site-url"),
```

Comment:
I don't see the point of passing `siteUrl` since the only place you use that also gets the `siteUrl` from the settings which should have the same value as `MetabaseSettings.get("site-url")` anyway.

https://github.com/metabase/metabase/blob/2556785498373fe796583a706519989a4e64f9a9/frontend/src/metabase/dashboard/components/PublicLinkPopover/DashboardPublicLinkPopover.tsx#L23-L28

What was the problem you encountered that you needed to pass `siteUrl` in the first place? It'd be great if we could leave the function API as is (only accepts a single argument `uuid`)